### PR TITLE
Little version of MVP backend

### PR DESCRIPTION
**Pull request by fizcamposancos**
A little version of MVP back-end with some endpoints, db connected with TypeORM, sessions saved into the db, guard for routes based on the sessions (only in the findall() method to check it).

[http://127.0.0.1:3000/](http://127.0.0.1:3000/) has login button to authenticate using the OAuth system of 42

Then a cookie is created and will be available for 5 minutes to access all users at [http://127.0.0.1:3000/api/v1/user](http://127.0.0.1:3000/api/v1/user). All the other endpoint are not protected, just to try.

This PR is not aimed to be merged, just to "play" with it and mock some services while there definite version is not available, ie, direct SQL database, not through TypeORM.

 No tests added yet.

### CODE REVIEW

File: transcendence-app/src/app.module.ts [hunk 1]
```diff
@@ -11,6 +14,16 @@ import { validate } from './config/env.validation';
       validate,
     }),
     OAuth42Module,
+    TypeOrmModule.forRoot({
+      type: 'postgres',
+      host: 'db',
+      port: 5432,
+      username: 'transcendence',
+      password: 'example',
```
Comment (gbudau) [inline transcendence-app/src/app.module.ts:22]: suggestion:  We could use the `ConfigService` here and add this info to a .env file or environment variables.

**Review by gbudau — APPROVED** [review]
Thank you and good job!

File: transcendence-app/src/app.module.ts (hunk 1 above)
Comment (fizcamposancos) [reply transcendence-app/src/app.module.ts:22]: Yes, good point. If we finally decide to use TypeORM we can do that or even we can call a config file to clean the app.module with `TypeOrmModule.forRoot(config.typeorm.ts)` and in that file add the configuration with environment variables as you mentioned.

**Review by vdedios — APPROVED** [review]
Nice work! I like to see a functional mvp with typeorm in action to get a perspective of how it works. Thanks :)

**Comment by Damrod** [comment]
If we finally decide to merge, we'll need to fix the merge conflicts. Could you please merge main into this branch and solve all of the merge conflicts, please?

File: transcendence-app/src/typeorm/session.entity.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Column, Entity, Index, PrimaryColumn } from 'typeorm';
+import { ISession } from 'connect-typeorm';
+
+@Entity({ name: 'sessions' })
+export class SessionEntity implements ISession {
+  @Index()
+  @Column('bigint')
+  expiredAt = Date.now();
```
[the remark this answers is not in the document]
Comment (fizcamposancos) [reply transcendence-app/src/typeorm/session.entity.ts:8]: It was just to set a Date value, no reason behind it

**Comment by fizcamposancos** [comment]
> 

Right now, database configuration is already done by Victor, sessions with Memcached and guards by George. The only difference remaning is TypeOrm. If we finally decide to implement it It would be easier to do that directly from the main branch and just close this PR.

File: transcendence-app/src/user/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  findOne(@Param('username') username: string) {
+    return this.userService.findOne(username);
+  }
+
+  @Get()
+  @UseGuards(RouteGuard)
+  findAll() {
+    return this.userService.findAll();
+  }
+
```
Comment (Damrod) [inline transcendence-app/src/user/user.controller.ts:28]: We decided to put 'provider' as a field in the UserDto. Why did you decide we don't want it anymore? I can think of some reasons, but I'd like to know your opinion.

File: transcendence-app/src/user/user.service.ts [hunk 4]
```diff
@@ -1,27 +1,45 @@
...
+    private readonly userRepository: Repository<UserEntity>,
+  ) {}
+
   private readonly users: UserDto[] = [];
 
-  private findOne(username: string) {
-    const user = this.users.find((user) => user.username === username);
-    return user;
+  async findOne(username: string) {
+    return await this.userRepository.findOne({ username: username });
```
Comment (Damrod) [inline transcendence-app/src/user/user.service.ts:17]: We could shortcircuit to a null as gbudau did. Why is null better than undefined? I don't know. I'll ask there.

**Review by Damrod — APPROVED** [review]

**Comment by fizcamposancos** [comment]
Just close it, for the time being

**Pull request closed without merging**